frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1842,6 +1953,49 @@ class SegmentsView extends React.Component<Props, State> {
     );
   }
 
+  getExpandSubgroupsItem(groupId: number) {
+    const children: Key[] = this.getSubGroupsAsTreeNodes(groupId);
+    const expandedGroupsSet = new Set(this.state.expandedGroupKeys);
+    const areAllChildrenExpanded = children.every((childNode) => expandedGroupsSet.has(childNode));
+    const isGroupItselfExpanded = expandedGroupsSet.has(getKeyForGroupId(groupId));
+    if (areAllChildrenExpanded && isGroupItselfExpanded) {
+      return null;
+    }
+    const allExpandedGroups = children.concat(this.state.expandedGroupKeys ?? []);
+    if (!isGroupItselfExpanded) allExpandedGroups.push(getKeyForGroupId(groupId));
+    return {
+      key: "expandAll",
+      disabled: !this.props.allowUpdate,
+      onClick: () => {
+        this.expandGroups(allExpandedGroups);
+        this.closeSegmentOrGroupDropdown();
+      },
+      icon: <ExpandAltOutlined />,
+      label: "Expand all subgroups",
+    };
+  }
+
+  getCollapseSubgroupsItem(groupId: number) {
+    const children = this.getSubGroupsAsTreeNodes(groupId);
+    const expandedKeySet = new Set(this.state.expandedGroupKeys);
+    const areAllChildrenCollapsed =
+      children.filter((childNode) => expandedKeySet.has(childNode)).length === 0;
+    const isGroupItselfCollapsed = !expandedKeySet.has(getKeyForGroupId(groupId));
+    if (areAllChildrenCollapsed || isGroupItselfCollapsed) {
+      return null;
+    }
+    return {
+      key: "collapseAll",
+      disabled: !this.props.allowUpdate,
```

Comment:
I wouldn't disable collapse and expand UI actions, because users might be interested in inspecting the hierarchy. it shouldn't be saved anyway and there will no be data loss when it's not saved.

Reply:
yes! this is a remnant of the old PR